js_modules/dagit/packages/core/src/runs/RunTimeline.tsx

```diff
@@ -169,16 +175,35 @@ interface TimelineSectionProps {
   range: [number, number];
   width: number;
   onToggle: (repoAddress: RepoAddress) => void;
+  onToggleAll: (expanded: boolean) => void;
 }
 
 const TimelineSection = (props: TimelineSectionProps) => {
-  const {expanded, onToggle, repoKey, isDuplicateRepoName, jobs, range, top, width} = props;
+  const {
+    expanded,
+    onToggle,
+    onToggleAll,
+    repoKey,
+    isDuplicateRepoName,
+    jobs,
+    range,
+    top,
+    width,
+  } = props;
   const repoAddress = repoAddressFromPath(repoKey);
   const repoName = repoAddress?.name || 'Unknown repo';
   const repoLocation = repoAddress?.location || 'Unknown location';
-  const onClick = React.useCallback(() => {
-    repoAddress && onToggle(repoAddress);
-  }, [onToggle, repoAddress]);
+
+  const onClick = React.useCallback(
+    (e: React.MouseEvent) => {
+      if (e.getModifierState('Shift')) {
```

Comment:
Hmm I'm not familiar with this API but I guess it's preferable to e.shiftKey? (Or maybe that's only available on keyboard events?) Will have to do some research!

Reply:
`shiftKey` is available on `MouseEvent` too, I just happened to use this instead. :) I believe it doesn't matter which one we use here, they should end up behaving the same.